cpp/daal/src/algorithms/dtrees/forest/oneapi/df_feature_type_helper_oneapi.i

```diff
@@ -107,20 +107,21 @@ services::Status IndexedFeaturesOneAPI<algorithmFPType>::FeatureEntry::allocBord
     return status;
 }
 
+template <typename algorithmFPType>
+size_t IndexedFeaturesOneAPI<algorithmFPType>::getRequiredMemSize(size_t nCols, size_t nRows)
+{
+    size_t requiredMem = sizeof(uint32_t) * (nCols + 1);
+
+    requiredMem += sizeof(uint32_t) * nRows * nCols; // data vs ftrs bin map table (_fullData)
```

Comment:
Why uint32_t is hardcoded? Do we need to make overflow checks here?

Reply:
introduced BinType, and added overflow check, thanks